Sources/Offline/OfflineManager.swift

```
@@ -0,0 +1,285 @@
+            if item?.state == .new {
+                // In case the item is still in the new state then something happened, shouldn't be an item with this state, remove it and create a new one.
+                try ContentManager.shared.removeItem(id: itemId)
+                item = nil
+            }
+            
+            if item == nil {
+                item = try ContentManager.shared.addItem(id: itemId, url: mediaSource.contentUrl!)
+            }
+        } catch {
+            PKLog.error("The item can't be added: " + error.localizedDescription)
+            callback(OfflineManagerError.itemCanNotBeAdded(message: error.localizedDescription), nil)
+            return
+        }
+
+        guard let dtgItem = item else {
+            PKLog.error("The item can't be added: ContentManager - Empty item returned.")
+            callback(OfflineManagerError.itemCanNotBeAdded(message: "ContentManager - Empty item returned."), nil)
+            return
+        }
+        
+        let assetInfo = AssetInfo(item: dtgItem)
+        
+        DispatchQueue.global().async {
+            do {
+                try ContentManager.shared.loadItemMetadata(id: itemId, options: options)
+                
+                PKLog.debug("Item Metadata Loaded")
+                callback(nil, assetInfo)
+            } catch {
+                DispatchQueue.main.async {
+                    PKLog.error("Load item metadata failed: " + error.localizedDescription)
+                    callback(OfflineManagerError.loadItemMetadataFailed(message: error.localizedDescription), nil)
+                }
+            }
+        }
+    }
+    
+    /**
+        Start or resume downloading the asset.
+    
+        * Parameters:
+            * assetId: The asset's id.
+    */
+    public func startAssetDownload(assetId: String) {
+        do {
+            try ContentManager.shared.startItem(id: assetId)
+        } catch {
+            PKLog.error(error.localizedDescription)
+        }
```

What about reporting the error to the app?
Same for other functions below. It seems that failure is ignored... ?

---

Fixed, please review the changes.